tests/internal/svcproc/svcproc.go

```diff
@@ -4,128 +4,89 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"os"
 	"os/exec"
-	"path/filepath"
-	"strings"
 	"syscall"
-	"time"
 
 	"go.uber.org/fx"
 
 	"go.autokitteh.dev/autokitteh/backend/svc"
 	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
 )
 
+const (
+	httpAddrFile = "ak_server_addr.txt"
+)
+
 type svcProc struct {
 	binPath string
-	ropts   svc.RunOptions
 	cfg     *svc.Config
-	addr    string
+	ropts   svc.RunOptions
 	cmd     *exec.Cmd
 	wait    chan svc.ShutdownSignal
 }
 
-func (s *svcProc) Start(ctx context.Context) error {
-	tmpDir, err := os.MkdirTemp("", "ak-svcproc-*")
-	if err != nil {
-		return fmt.Errorf("mkdirtemp: %w", err)
+func NewSvcProc(binPath string, cfg *svc.Config, ropts svc.RunOptions) (svc.Service, error) {
+	if ropts.TemporalClient != nil {
+		return nil, sdkerrors.ErrNotImplemented
 	}
 
-	defer func() {
-		if err := os.RemoveAll(tmpDir); err != nil {
-			fmt.Fprintf(os.Stderr, "rm(%q): %v\n", tmpDir, err)
-		}
-	}()
+	return &svcProc{
+		binPath: binPath,
+		cfg:     cfg,
+		ropts:   ropts,
+	}, nil
+}
+
+func (s *svcProc) Start(ctx context.Context) error {
+	// TODO: Pass user configuration to executable.
+	s.cmd = exec.Command(
+		s.binPath, "up",
+		"--mode", string(s.ropts.Mode),
 
-	addrFilePath := filepath.Join(tmpDir, "addr")
-	readyFilePath := filepath.Join(tmpDir, "ready")
+		"--config", "db.type=sqlite",
+		"--config", "db.dsn=file:autokitteh.sqlite", // In the test's temporary directory.
 
-	// TODO: pass user configuration to executable.
-	cmd := exec.Command(
-		s.binPath,
 		"--config", "http.addr=:0",
-		"--config", fmt.Sprintf("http.addr_filename=%s", addrFilePath),
-		"up",
-		"-m", string(s.ropts.Mode),
-		"-r", readyFilePath,
+		"--config", "http.addr_filename="+httpAddrFile, // In the test's temporary directory.
 	)
-	// use same system group to kill ak + all children (temporal, etc.)
-	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	// Use same system group to kill ak + all children (temporal, etc.)
+	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	if err := cmd.Start(); err != nil {
-		return fmt.Errorf("start: %w", err)
+	if err := s.cmd.Start(); err != nil {
+		return fmt.Errorf("start subprocess: %w", err)
 	}
 
-	wait := make(chan fx.ShutdownSignal, 1)
+	s.wait = make(chan fx.ShutdownSignal, 1)
 	go func() {
-		var rc int
+		var ec int
```

Comment:
dude nobody knows what `ec` is by just looking at it

Reply:
See line 66: `ec = exitError.ExitCode()`, or line 73: `{ExitCode: ec}`.

"rc" is the acronym without explanation. :)

Reply:
ec -> exitCode? or ret?
Short var names help to minimize code clutter and thus increasing readability.
In this specific case IMO variable name `ec` reduces readability. Let's use meaningful names? e.g. `exitCode` or at least `ret` ?

Reply:
makes sense, I'll use `exitCode` for consistency